angr/analyses/disassembly.py

```diff
@@ -927,10 +964,132 @@ def parse_block(self, block: BlockNode) -> None:
+                buf.append(pad + item.render(formatting)[0])
+            elif isinstance(item, Comment):
+                comment = item
+            elif isinstance(item, Instruction):
+                a2ln[item.addr].append(len(buf))
+                lines = []
+
+                # Chop instruction bytes into line segments
+                p, insn_bytes = 0, []
+                while show_bytes and p < len(item.insn.bytes):
+                    s = item.insn.bytes[p:p+min(len(item.insn.bytes)-p, max_bytes_per_line)]
+                    p += len(s)
+                    insn_bytes.append(s)
+
+                # Format the instruction's address, bytes, disassembly, and comment
+                s_plain = format_address(item.addr, False)
+                s = format_address(item.addr)
+                if show_bytes:
+                    bytes_column = len(s_plain)
+                    s_plain += format_bytes(insn_bytes[0], False)
+                    s += format_bytes(insn_bytes[0])
+                s_plain += item.render()[0]
+                s += item.render(formatting)[0]
+                if comment is not None:
+                    comment_column = len(s_plain)
+                    s += format_comment(comment.text[0])
+                lines.append(s)
+
+                # Add additional lines of instruction bytes
+                for i in range(1, len(insn_bytes)):
+                    lines.append(' ' * bytes_column + format_bytes(insn_bytes[i]))
+
+                # Add additional lines of comments
+                if comment is not None:
+                    for i in range(1, len(comment.text)):
+                        if len(lines) <= i:
+                            lines.append(' ' * comment_column)
+                        lines[i] += format_comment(comment.text[i])
+                    comment = None
+
+                buf.extend(lines)
+            else:
+                buf.append(item.render(formatting))
+
+        if self._graph is not None and show_edges and len(buf) > 0:
+            edges_by_line = set()
+            for edge in self._graph.edges.items():
+                from_block, to_block = edge[0]
+                if to_block.addr != from_block.addr + from_block.size:
+                    from_addr = edge[1]['ins_addr']
+                    to_addr = to_block.addr
+                    if not (from_addr in a2ln and to_addr in a2ln):
+                        continue
+                    for f in a2ln[from_addr]:
+                        for t in a2ln[to_addr]:
+                            edges_by_line.add((f, t))
+
+            # Render block edges, to a reference buffer for tracking and output buffer for display
+            edge_buf = ['' for _ in buf]
+            ref_buf = ['' for _ in buf]
+            for f, t in sorted(edges_by_line, key=lambda e: abs(e[0]-e[1])):
+                add_edge_to_buffer(edge_buf, ref_buf, f, t, lambda s: ansi_color(s, col('edge')), ascii_only=ascii_only)
```

Comment:
if `col('edge')` does not change in the loop body, can we put it outside the loop to prevent from calling it multiple times?

Reply:
Eventually I do want to have edge classification+coloring here but it's unchanging for now, fixed